### PR TITLE
docs: move /var/log/README to a tmpfiles.d symlink

### PR DESCRIPTION
**Pull request by lucab**
This moves the /var/log/README content out of /var and into the
docs location, replacing the previous file with a symlink
created through a tmpfiles.d entry.

### CODE REVIEW

File: tmpfiles.d/systemd.conf.in [hunk 1]
```diff
@@ -45,9 +45,10 @@ a+ /run/log/journal/%m/*.journal* - - - - group:wheel:r--
 {% endif %}
 {% endif %}
 
-z /var/log/journal 2755 root systemd-journal - -
-z /var/log/journal/%m 2755 root systemd-journal - -
-z /var/log/journal/%m/system.journal 0640 root systemd-journal - -
+L+ /var/log/README     - - - -
```
Comment (mbiebl) [inline tmpfiles.d/systemd.conf.in]: I'd prefer, if this was a separate .tmpfile.

**Comment by mbiebl** [comment]
Isn't factory content supposed to be copied instead of symlinked?

**Comment by keszybz** [comment]
> Isn't factory content supposed to be copied instead of symlinked?

I don't think it matters in this case. The file is not supposed to be edited.

File: tmpfiles.d/systemd.conf.in (hunk 1 above)
Comment (keszybz) [reply tmpfiles.d/systemd.conf.in]: In the removed meson file, the file was conditionalized as `conf.get('HAVE_SYSV_COMPAT') == 1 and get_option('create-log-dirs')`. Shouldn't the same logic be used here?
```
conf.set10('CREATE_LOG_DIRS', get_option('create-log-dirs'))

{% if HAVE_SYSV_COMPAT and CREATE_LOG_DIRS %}
L+ /var/log/README  - - - -
{% endif %}
```

> I'd prefer, if this was a separate .tmpfile.

Is the 'create-log-dirs' option not enough?

File: meson.build [hunk 2]
```diff
@@ -3514,10 +3514,10 @@ subdir('man')
 subdir('shell-completion/bash')
 subdir('shell-completion/zsh')
 subdir('docs/sysvinit')
-subdir('docs/var-log')
 
-install_subdir('factory/etc',
-               install_dir : factorydir)
+install_subdir('factory',
+               install_dir : factorydir,
+               strip_directory : true)
```
Comment (keszybz) [inline meson.build]: Hmm, so we lose the conditionalization on `conf.get('HAVE_SYSV_COMPAT')`…

File: tmpfiles.d/systemd.conf.in (hunk 1 above)
Comment (mbiebl) [reply tmpfiles.d/systemd.conf.in]: > > I'd prefer, if this was a separate .tmpfile.
> 
> Is the 'create-log-dirs' option not enough?

I guess that would be fine as well

Comment (mbiebl) [reply tmpfiles.d/systemd.conf.in]: or a separate option for all those READMEs sprinkled across the system:
```
./var/log/README
./etc/init.d/README
./lib/modprobe.d/README
./lib/udev/hwdb.d/README
./lib/udev/rules.d/README
./usr/lib/tmpfiles.d/README
./usr/lib/sysusers.d/README
./usr/lib/sysctl.d/README
```

**Comment by lucab** [comment]
I went for the "simpler is better" approach here, but I can happily steer this in a different direction wrt. conditionals / locations / copying.
Just let me know when there is consensus on where to go instead.

My real-word itch here is "remove the last contentful file from /var" so I'd be happier to keep this PR scoped to that (i.e. without venturing into larger /etc or /usr changes).

File: tmpfiles.d/systemd.conf.in (hunk 1 above)
Comment (keszybz) [reply tmpfiles.d/systemd.conf.in]: > or a separate option for all those READMEs sprinkled across the system:

I think this is a completely separate subject. Those README files are in-place docs about a given directory.

**Comment by keszybz** [comment]
From the point of view of a person who doesn't want sysv compat (empty `-Dsysvinit-path= -Dsysvrcnd-path=`), adding this file seems like a regression.

Ideally, we would keep the conditionalization, and only install `/usr/share/factory/var/log/README` and have the `L+` line if `conf.get('HAVE_SYSV_COMPAT') == 1 and get_option('create-log-dirs')`. This is a bit more complex than the initial patch, but we don't want to reintroduce obsolete stuff.

Also, could you do `/etc/rc.d/README` too while at it?

**Comment by poettering** [comment]
hmm, is factory really the right place for this? shouldn't this be placed in docdir or so?

I'd use the factory only for stuff that is a "default" without which the boot otherwise fails if unpopulated. I am not sure the READMEs really qualify for that. They are a nicety to improve transition from old systems to the new. But if you buy into the logic of booting with unpopulated /var and /etc then you are already super far away from those traditional systems.

Hence, I'd just make this stuff that we install by default, but would never reconstruct if missing.

**Comment by keszybz** [comment]
> hmm, is factory really the right place for this? shouldn't this be placed in docdir or so?

Hmm, maybe? If we installed this as /usr/share/doc/systemd/README.sysvinit (or something like that) and then created the symlink this should work.

**Comment by keszybz** [comment]
> `L+`

I think this should be just `L`. We don't have good reason to forcibly replace the file.

**Comment by poettering** [comment]
I'd prefer this to be in docdir, instead of the factory dir, at least. and I guess it's OK to keep as long as it can be turned off via meson, and is all placed in one tmpfiles snippet, so it's easy to remove

**Comment by lucab** [comment]
Thanks for the feedback, I pushed another revision of this. Bunch of notes from my side:
 * there is already a `tmpfiles.d/legacy.conf` which is both a dedicated separate snippet for traditional systems and conditional on meson sysv options. It seemed to fit your requests, so I used that instead of adding another separate file. Let me know in case I misjudged this.
 * I had a look at moving the `/etc/rc.d` file too but got stuck fighting meson in order to get relative path from-docdir-to-sysvdir. I eventually gave up and left that to TODO. To be honest I wasn't originally planning to tackle that here, just please don't hate me :)

**Comment by lucab** [comment]
Gentle ping for another review pass.

**Comment by keszybz** [comment]
LGTM. I removed /var/log/README and it gets recreated as expected.

**Pull request merged**